packages/gestalt-datepicker/src/DatePicker.js

```diff
@@ -124,20 +146,20 @@ const DatePickerWithForwardRef: React$AbstractComponent<Props, HTMLDivElement> =
         dateFormat={format}
         dayClassName={() => classnames(styles['react-datepicker__days'])}
         disabled={disabled}
-        endDate={rangeEndDate}
-        excludeDates={excludeDates && [...excludeDates]}
+        endDate={rangeEndDate && getDisplayDate(rangeEndDate, timezone)}
+        excludeDates={excludeDates && excludeDates.map((date) => getDisplayDate(date, timezone))}
         highlightDates={initRangeHighlight ? [initRangeHighlight] : []}
```

Comment:
`initRangeHighlight` this is a date as well

Reply:
line 101 you might had to add function as well

`setInitRangeHighlight(rangeStartDate || rangeEndDate);`